### PR TITLE
Refactor CreateSecret

### LINKED ISSUE #11437: test/utils.go: Inline CreateSecret
**Is your feature request related to a problem? Please describe**:

The function `CreateSecret` in tests/utils.go could be inlined, since it does not provide much additional value. Also since this is test code, assertions should be used instead of panicking on errors.

**Describe the solution you'd like**:

Inline `CreateSecret` into the places where it is used.

### PR DESCRIPTION
**Pull request by sarthaksarthak9**
Refactor createScret() by inlining the logic where function is used

### What this PR does
Before this PR:

After this PR:

Fixes #11437 

### Why we need it and why it was done in this way
The following tradeoffs were made:

The following alternatives were considered:

Links to places where the discussion took place: 

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

### Release note

```release-note
None
```

### CODE REVIEW

File: tests/config_test.go [hunk 1]
```diff
@@ -181,7 +187,16 @@ var _ = Describe("[rfe_id:899][crit:medium][vendor:cnv-qe@redhat.com][level:comp
 					"user":     "admin",
 					"password": "redhat",
 				}
-				tests.CreateSecret(secretName, testsuite.GetTestNamespace(nil), data)
```
Comment (xpivarc) [inline tests/config_test.go:184]: Is the `CreateSecret` used anywhere now?

File: tests/config_test.go [hunk 2]
```diff
@@ -181,7 +187,16 @@ var _ = Describe("[rfe_id:899][crit:medium][vendor:cnv-qe@redhat.com][level:comp
...
+
+				virtCli := kubevirt.Client()
+
+				_, err := virtCli.CoreV1().Secrets(testsuite.GetTestNamespace(nil)).Create(context.Background(), &k8sv1.Secret{
+					ObjectMeta: metav1.ObjectMeta{Name: secretName},
+					StringData: data,
+				}, metav1.CreateOptions{})
+				if !errors.IsAlreadyExists(err) {
```
Comment (xpivarc) [inline tests/config_test.go:197]: We should expect no error `Expect(err).To(Succeed)`

**Comment by victortoso** [comment]
I did try to compile and run locally your branch. It is failing with:

```
tests/virtiofs/config.go:131:102: undefined: k8sv1
```

Note that you should use `make generate` for the Bazel related changes. You can run `make generate` and add the changes to your commit. Current `make generate` ouputs:

```diff
diff --git a/tests/virtiofs/BUILD.bazel b/tests/virtiofs/BUILD.bazel
index e01e64408f..5718cc0a1f 100644
--- a/tests/virtiofs/BUILD.bazel
+++ b/tests/virtiofs/BUILD.bazel
@@ -33,6 +33,7 @@ go_library(
         "//vendor/github.com/onsi/ginkgo/v2:go_default_library",
         "//vendor/github.com/onsi/gomega:go_default_library",
         "//vendor/k8s.io/api/core/v1:go_default_library",
+        "//vendor/k8s.io/apimachinery/pkg/api/errors:go_default_library",
         "//vendor/k8s.io/apimachinery/pkg/api/resource:go_default_library",
         "//vendor/k8s.io/apimachinery/pkg/apis/meta/v1:go_default_library",
         "//vendor/kubevirt.io/containerized-data-importer-api/pkg/apis/core/v1beta1:go_default_library",
diff --git a/tests/virtiofs/config.go b/tests/virtiofs/config.go
index 63fce193b9..143832140a 100644
--- a/tests/virtiofs/config.go
+++ b/tests/virtiofs/config.go
@@ -25,6 +25,7 @@ import (

        "k8s.io/apimachinery/pkg/api/errors"
        v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
        virtconfig "kubevirt.io/kubevirt/pkg/virt-config"
        "kubevirt.io/kubevirt/tests/decorators"
        "kubevirt.io/kubevirt/tests/framework/checks"
```

File: tests/config_test.go [hunk 3]
```diff
@@ -240,7 +254,18 @@ var _ = Describe("[rfe_id:899][crit:medium][vendor:cnv-qe@redhat.com][level:comp
...
+						"option": "value",
+					}
+					virtCli := kubevirt.Client()
+
+					_, err := virtCli.CoreV1().Secrets(testsuite.GetTestNamespace(nil)).Create(context.Background(), &k8sv1.Secret{
+						ObjectMeta: metav1.ObjectMeta{Name: name},
+						StringData: data,
+					}, metav1.CreateOptions{})
+					if !errors.IsAlreadyExists(err) {
+						util2.PanicOnError(err)
```
Comment (alicefr) [inline tests/config_test.go:267]: @sarthaksarthak9 please avoid to panic and include an assertion as it is reported in the https://github.com/kubevirt/kubevirt/issues/11437 description

**Comment by alicefr** [comment]
@sarthaksarthak9 is it now the function `CreateSecret` used somewhere?

**Comment by avlitman** [comment]
@alicefr Can you review this pr again?

**Comment by alicefr** [comment]
@sarthaksarthak9 Please, remove the merge commit, we need to have a clean github history.

File: tests/config_test.go [hunk 4]
```diff
@@ -201,6 +204,15 @@ var _ = Describe("[rfe_id:899][crit:medium][vendor:cnv-qe@redhat.com][level:comp
...
 				}
+
+				virtCli := kubevirt.Client()
+
+				_, err := virtCli.CoreV1().Secrets(testsuite.GetTestNamespace(nil)).Create(context.Background(), &k8sv1.Secret{
+					ObjectMeta: metav1.ObjectMeta{Name: secretName},
+					StringData: data,
+				}, metav1.CreateOptions{})
+				Expect(err).To(Succeed())
+
```
Comment (alicefr) [inline tests/config_test.go:215]: nit: extra empty space

File: tests/operator/operator.go [hunk 5]
```diff
@@ -247,7 +247,18 @@ var _ = Describe("[Serial][sig-operator]Operator", Serial, decorators.SigOperato
 					Expect(err).ToNot(HaveOccurred())
 				}
 
-				vmi := libvmifact.NewCirros(
+				tests.CreateConfigMap(configMapName, testsuite.GetTestNamespace(nil), config_data)
+				tests.CreateSecret(secretName, testsuite.GetTestNamespace(nil), secret_data)
```
Comment (alicefr) [inline tests/operator/operator.go:251]: Aren't we suppose to get rid of this function?

**Comment by alicefr** [comment]
@sarthaksarthak9 the goal of this PR should be to remove the CreateSecret function, but in your code it is still there

File: tests/operator/operator.go (hunk 5 above)
Comment (sarthaksarthak9) [reply tests/operator/operator.go:251]: sorry I miss this

**Comment by sarthaksarthak9** [comment]
@alicefr I checked the main branch, ig this issue is resolved

**Pull request closed without merging**